packages/aws-cdk/lib/cli-arguments.ts

```diff
@@ -143,14 +143,14 @@ export interface GlobalOptions {
   /**
    * Add contextual string parameter (KEY=VALUE)
    *
-   * @default - undefined
+   * @default - []
```

Comment:
yep. I changed the default to `[]` because providing `undefined` was also causing problems. elsewhere we are expecting the `context` field to at least exist, which it doesn't if we default to `undefined`. I know why this is happening, but still thinking through what the best path forward is.

Reply:
this PR is not meant to change any functionality. since yargs normally defaults to `undefined`, even for arrays, I will update this PR to reflect that and fix the unit tests that are failing in that case.

Reply:
ok i got to the bottom of this and essentially this bug in yargs is the problem: https://github.com/yargs/yargs/issues/2443

i'm back to defaulting arrays as `[]`, except for notification-arns which should be defaulted to `undefined`, but right now that leads to notification-arns being `[undefined]`.

going forward, it should be the case that unless otherwise specified, arrays are defaulted as `[]`